guides/source/deploy_performance_tuning.md

```diff
@@ -0,0 +1,146 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**
+
+Tuning Performance for Deployment
+=================================
+
+This guide covers performance and concurrency configuration for deploying your production Ruby on Rails application.
+
+After reading this guide, you will know:
+
+* Whether to use Puma, the default application server
+* How to configure important performance settings for Puma
+
+More information about how to configure your application can be found in the [Configuration Guide](configuring.html).
+
+--------------------------------------------------------------------------------
+
+Choosing an Application Server
+------------------------------
+
+An application server uses a particular concurrency method. For example Unicorn uses processes, Puma is hybrid process- and thread-based concurrency, Thin uses EventMachine and Falcon uses Ruby Fibers.
+
+A full discussion of Ruby's concurrency methods is beyond the scope of this document. If you want to use a method other than processes or threads, you will need to use a different application server. There are other features are only available with another server, such as Pitchfork's preforking.
+
+The most common application servers are used by removing the Puma gem from your Gemfile and including the gem for the server. Consult the appropriate application server documentation for details.
+
+Ruby Concurrency
+----------------
+
+Ruby has many kinds of concurrency. Puma supports a hybrid process-based and thread-based concurrency model.
+
+This guide is for [CRuby](https://ruby-lang.org), the original implementation of Ruby. If you're using a Ruby without a GVL such as JRuby or TruffleRuby, the configuration will be very different. If needed, check other sources specific to your Ruby implementation.
+
+### Process-Based Concurrency
+
+Puma calls its multi-process concurrency "clustered mode". In this mode it forks new worker processes from a master process and each one separately processes requests. Each worker is a fully-capable Ruby process, doing everything Ruby normally does.
+
+Creating new processes uses a lot of memory. Each worker contains all data from the master process. Ruby uses [copy-on-write memory](https://en.wikipedia.org/wiki/Copy-on-write) to avoid duplicating most master-process data that doesn't change. But process-based workers often use a lot of memory, especially long-running workers.
+
+Processes are resilient. Killing a single process doesn't affect other processes much. They are also slower to start up than threads. Loading your application in the master thread instead of the workers is called preloading. It helps startup time, but may increase memory use.
```

Comment:
```suggestion
Processes are resilient. Killing or crashing a single process doesn't affect other processes at all. Loading your application in the master process instead of the workers is called preloading. It helps reduce memory usage by increasing the amount of memory that can be shared with the parent via Copy-on-Write.
```

Reply:
Incorporated with minor tweaks.